### PR TITLE
Fix: Contextual menu focus with custom UI

### PR DESCRIPTION
**Pull request by micahgodbolt**
Looking deeper into focus functionality in the Contextualmenu, I realized that the original non queued approach was actually correct (each nested menu was its own contextual menu with its own context). The problem ended up being "last focused elements" either being switch to a child element (when focus returns TO a menu), or to the body (when the old menu closes). Filtering out those two cases allows the behavior to work as expected in 3+ menu depths, as well as custom UI like a textfield

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/97e3ebb627a8a24f805454fa0220f7e2827d3107'>97e3ebb627a8a24f805454fa0220f7e2827d3107</a></sub>
<div id="bundleSizeComment9423"/>

**Review by smhigley — APPROVED** [review]

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>BaseButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/BaseButton-mount_base.html">1752</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/BaseButton-mount.html">1147</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/BaseButton-mount.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>BaseButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/BaseButton-mount_base.html">1752</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/BaseButton-mount.html">1147</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/BaseButton-mount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>Breadcrumb</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/Breadcrumb-mount_base.html">2936</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/Breadcrumb-mount.html">2926</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Checkbox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/Checkbox-mount_base.html">2636</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/Checkbox-mount.html">2692</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>CheckboxBase</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/CheckboxBase-mount_base.html">2375</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/CheckboxBase-mount.html">2381</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ChoiceGroup</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/ChoiceGroup-mount_base.html">4500</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/ChoiceGroup-mount.html">4442</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ComboBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/ComboBox-mount_base.html">1317</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/ComboBox-mount.html">1325</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>CommandBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/CommandBar-mount_base.html">9667</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/CommandBar-mount.html">9628</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>ContextualMenu</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/ContextualMenu-mount_base.html">11017</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/ContextualMenu-mount.html">10986</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DefaultButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/DefaultButton-mount_base.html">1434</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/DefaultButton-mount.html">1397</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRow</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/DetailsRow-mount_base.html">3736</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/DetailsRow-mount.html">3675</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowFast</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/DetailsRowFast-mount_base.html">3647</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/DetailsRowFast-mount.html">3626</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowNoStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/DetailsRowNoStyles-mount_base.html">3526</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/DetailsRowNoStyles-mount.html">3504</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Dialog</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/Dialog-mount_base.html">3154</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/Dialog-mount.html">3132</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DocumentCardTitle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/DocumentCardTitle-mount_base.html">589</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/DocumentCardTitle-mount.html">593</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Dropdown</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/Dropdown-mount_base.html">3269</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/Dropdown-mount.html">3228</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusTrapZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/FocusTrapZone-mount_base.html">1988</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/FocusTrapZone-mount.html">1946</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/FocusZone-mount_base.html">1971</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/FocusZone-mount.html">1989</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedList</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/GroupedList-mount_base.html">53553</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/GroupedList-mount.html">60199</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedList</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/GroupedList-virtual-rerender_base.html">24912</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/GroupedList-virtual-rerender.html">25385</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedList</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/GroupedList-virtual-rerender-with-unmount_base.html">98692</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/GroupedList-virtual-rerender-with-unmount.html">93952</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedListV2</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/GroupedListV2-mount_base.html">591</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/GroupedListV2-mount.html">562</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedListV2</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/GroupedListV2-virtual-rerender_base.html">535</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/GroupedListV2-virtual-rerender.html">539</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedListV2</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/GroupedListV2-virtual-rerender-with-unmount_base.html">569</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/GroupedListV2-virtual-rerender-with-unmount.html">567</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>IconButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/IconButton-mount_base.html">1931</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/IconButton-mount.html">1908</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Label</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/Label-mount_base.html">752</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/Label-mount.html">731</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Layer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/Layer-mount_base.html">4268</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/Layer-mount.html">4354</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Link</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/Link-mount_base.html">826</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/Link-mount.html">821</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MenuButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/MenuButton-mount_base.html">1694</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/MenuButton-mount.html">1700</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MessageBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/MessageBar-mount_base.html">2356</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/MessageBar-mount.html">2328</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Nav</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/Nav-mount_base.html">3313</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/Nav-mount.html">3274</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>OverflowSet</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/OverflowSet-mount_base.html">1378</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/OverflowSet-mount.html">1375</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Panel</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/Panel-mount_base.html">2543</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/Panel-mount.html">2564</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/Persona-mount_base.html">1283</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/Persona-mount.html">1286</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Pivot</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/Pivot-mount_base.html">1656</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/Pivot-mount.html">1655</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>PrimaryButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/PrimaryButton-mount_base.html">1519</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/PrimaryButton-mount.html">1541</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Rating</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/Rating-mount_base.html">7064</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/Rating-mount.html">7019</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SearchBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/SearchBox-mount_base.html">1509</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/SearchBox-mount.html">1506</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Shimmer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/Shimmer-mount_base.html">2868</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/Shimmer-mount.html">2937</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Slider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/Slider-mount_base.html">2113</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/Slider-mount.html">2133</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/SpinButton-mount_base.html">4731</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/SpinButton-mount.html">4734</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Spinner</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/Spinner-mount_base.html">828</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/Spinner-mount.html">797</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SplitButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/SplitButton-mount_base.html">3097</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/SplitButton-mount.html">3146</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Stack</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/Stack-mount_base.html">862</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/Stack-mount.html">858</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithIntrinsicChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/StackWithIntrinsicChildren-mount_base.html">2368</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/StackWithIntrinsicChildren-mount.html">2383</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithTextChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/StackWithTextChildren-mount_base.html">4808</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/StackWithTextChildren-mount.html">4988</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchColorPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/SwatchColorPicker-mount_base.html">10531</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/SwatchColorPicker-mount.html">10579</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TagPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/TagPicker-mount_base.html">2700</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/TagPicker-mount.html">2653</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TeachingBubble</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/TeachingBubble-mount_base.html">88400</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/TeachingBubble-mount.html">87863</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Text</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/Text-mount_base.html">786</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/Text-mount.html">801</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TextField</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/TextField-mount_base.html">1604</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/TextField-mount.html">1601</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/ThemeProvider-mount_base.html">1509</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/ThemeProvider-mount.html">1567</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/ThemeProvider-virtual-rerender_base.html">1092</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/ThemeProvider-virtual-rerender.html">1078</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/ThemeProvider-virtual-rerender-with-unmount_base.html">2186</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/ThemeProvider-virtual-rerender-with-unmount.html">2183</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Toggle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/Toggle-mount_base.html">1101</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/Toggle-mount.html">1128</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>buttonNative</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/buttonNative-mount_base.html">545</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25063/perf-test/buttonNative-mount.html">557</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9423"/>

File: packages/react/src/components/ContextualMenu/ContextualMenu.base.tsx [hunk 1]
```diff
@@ -252,8 +250,12 @@ function useShouldUpdateFocusOnMouseMove({ delayUpdateFocusOnHover, hidden }: IC
   return [shouldUpdateFocusOnMouseEvent, gotMouseMove, onMenuFocusCapture] as const;
 }
 
-function usePreviousActiveElement({ hidden, onRestoreFocus }: IContextualMenuProps, targetWindow: Window | undefined) {
-  const previousElementsQueue = React.useRef<HTMLElement[]>([]);
+function usePreviousActiveElement(
+  { hidden, onRestoreFocus }: IContextualMenuProps,
+  targetWindow: Window | undefined,
+  hostElement: any,
```
Comment (micahgodbolt) [inline packages/react/src/components/ContextualMenu/ContextualMenu.base.tsx:256]: Need access to the host element to know if the newly focused element is within that host (like focus moving to a searchbox)

File: packages/react/src/components/ContextualMenu/ContextualMenu.base.tsx [hunk 2]
```diff
@@ -252,8 +250,12 @@ function useShouldUpdateFocusOnMouseMove({ delayUpdateFocusOnHover, hidden }: IC
...
+) {
+  const previousActiveElement = React.useRef<undefined | HTMLElement>();
```
Comment (micahgodbolt) [inline packages/react/src/components/ContextualMenu/ContextualMenu.base.tsx:258]: this and 268 is just returning to the original approach

**Review by smhigley — APPROVED** [review]

**Pull request merged**